docs/modules/ROOT/pages/configuration-store.adoc

```diff
@@ -0,0 +1,172 @@
+= Universal Configuration Store
+
+== Overview
+The Universal Configuration Store is a tool that enables the various configurations for a project to be centrally defined and managed. 
+It then provides a standardized way of accessing them, allowing the environment specific configurations to be dynamically provided to 
+their respective resources within the project at runtime.
+
+=== Setup
+To setup the configuration store, one must first create properties files for storing the configurations. Each individual configuration is 
+represented as a property key value pair within the file. These files must then be made accessible to the configuration store kubernetes 
+resource for ingestion. See the applicable sections below for more details on defining configurations as properties, creating environment 
+specific property overrides, mounting the properties, and specifying the configuration store backend storage type.
+
+==== Defining Configuration Properties
+Properties represent key values pairs for storing the project configurations. The key values pairs are stored as string types using 
+the standard `.properties` file format. Each property then has an associated group name determined by the name of the file. Properties
+are then accessed using a combination of their group name and property name. Accessing the properties is covered in more detail in the 
+<<Usage>> section below.
+
+Here is an example of a base properties file named `messaging.properties`:
+[source,properties]
+----
+connector=smallrye-kafka
+topic=baseTopic
+----
+This file defines two properties, `connector` and `topic`, each with the group name `messaging`.
+
+===== Environment Specific Overrides
+In addition to the base properties file introduced above, one can optionally specify a secondary properties file that houses environment
+specific configurations. These configurations will then used to override their respective base configurations.
+
+Continuing with the example above, suppose one defines an additional properties file named `messaging.properties` like the following:
+[source,properties]
+----
+topic=ciTopic
+auth-token=auth-secret
+----
+
+Then when these two properties files are reconciled within the store, it results in a configuration equivalent to the following:
+[source,properties]
+----
+connector=smallrye-kafka
+topic=ciTopic
+auth-token=auth-secret
+----
+
+NOTE: The `topic` property is overwritten when the environment specific value `ciTopic`.
+
+==== Mounting Properties
```

Comment:
S: I think this would be more consumable if the over-arching section was "Deploying" and we frame everything in that context.  We could have a quick intro paragraph that talks about/links the Helm chart documentation, and what the key goals are of the deployment setup (making the configuration properties files available to the service and choosing the backing store)

Reply:
Refactored